### ISSUE TITLE
Understanding for new commers

### ISSUE THREAD

**Issue by Ricocotam** [harvardnlp/namedtensor#94]
Hi, 
I'm new to this and I'd like it to be standard ASAP. But first, I have to understand a bit more about how this works.
In the Readme, there's a few example I don't understand, and thus lack explanation I guess. Here they are :

```python
x = ntorch.randn(10, 10, 20, names=("batch", "h", "w"))
select = ntorch.tensor([1, 4, 5], names=("rows",))
y = x[{"h": select}] 
# y shape ("batch", "rows", "w")
```
What does it do when you write `ntorch.tensor([1, 4, 5], names=("rows",))` ? How are non named dimensions named ? I don't undersand the line `y = x[{"h": select}] `, could you help ?

Why removing the expand ? There's a lot of practical case where you need expanding for some broadcasting. Then, why not removing repeat ?

I'll come later with other questions about the implementation (I have to try a few things first).

Thanks for your answers

**Comment by srush** [comment]
Good questions. 

> What does it do when you write ntorch.tensor([1, 4, 5], names=("rows",)) ?

This creates a one-dimensional tensor [1, 4, 5] with name "rows". (That name is confusing, I'll change it)

> How are non named dimensions named ?

All dimensions must be named. 

> I don't undersand the line y = x[{"h": select}] , could you help ?

That is equivalent to `y = x[:, select]` in normal torch. Instead of having to no the position of the "h" axis, you give the name. 

> Why removing the expand ?

The idea was that broadcasting doesn't need this any more, extra dimensions get added automatically. However there are some non-broadcasting use cases that may make us add it back.
